src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -225,7 +234,7 @@ public class RSessionProvider : IRSessionProvider {
             var oldBroker = _brokerProxy.Set(brokerClient);
             try {
                 BrokerChanging?.Invoke(this, EventArgs.Empty);
-                await SwitchBrokerAsync(oldBroker, cancellationToken, lockToken.Reentrancy);
+                await SwitchBrokerAsync(cts.Token, lockToken.Reentrancy);
```

Comment:
yes, that's true.

---
In reply to: [83921016](https://github.com/Microsoft/RTVS/pull/2529#discussion_r83921016) [](ancestors = 83921016)